test/networkd-test.py

```diff
@@ -96,6 +96,46 @@ def write_network_dropin(self, unit_name, dropin_name, contents):
             dropin.write(contents)
         self.addCleanup(os.remove, dropin_path)
 
+    def assert_link_states(self, **kwargs):
+        """Match networkctl link states to the given ones.
+
+        Each keyword argument should be the name of a network interface
+        with its expected value of the "SETUP" column in output from
+        networkctl.  The interfaces have five seconds to come online
+        before the check is performed.  Every specified interface must
+        be present in the output, and any other interfaces found in the
+        output are ignored.
+
+        A special interface state "managed" is supported, which matches
+        any value in the "SETUP" column other than "unmanaged".
+        """
+        if not kwargs:
+            return
+        succeeded = dict.fromkeys(kwargs.keys(), False)
```

Comment:
This and the tests below could be simplified a bit by replacing this with `interfaces = set(kwargs)`...